### PR TITLE
DOC: readme and contributing doc cleanup

### LINKED ISSUE #850: Installation of development version
Our README instructions to install the development version might mislead a new developer to clone biocore's repo instead of forking and modifying that, plus they use ssh instead of https cloning, which is another roadblock for a would-be contributor. We just need to point readers to CONTRIBUTING.md, which has the right steps.

### PR DESCRIPTION
**Pull request by jairideout**
Took a pass through README.rst and CONTRIBUTING.md, updating existing text to be more succinct and adding new info relevant to scikit-bio developers. Instructions for installing the development version of scikit-bio are now included in CONTRIBUTING.md instead of README.rst.

Fixes #850. Addresses part of #867.

@gregcaporaso @ebolyen @Jorge-C can you please review?

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -5,103 +5,67 @@
...
 
--  `QIIME <http://qiime.org/>`__
--  `Emperor <http://biocore.github.io/emperor/>`__
--  `An Introduction to Applied
-   Bioinformatics <http://caporasolab.us/An-Introduction-To-Applied-Bioinformatics/>`__
--  `tax2tree <https://github.com/biocore/tax2tree>`__
+- `QIIME <http://qiime.org/>`__
+- `Emperor <http://biocore.github.io/emperor/>`__
+- `An Introduction to Applied
+  Bioinformatics <http://applied-bioinformatics.org>`__
```
Comment (gregcaporaso) [inline README.rst]: Can you use the new URL for IAB:

readIAB.org

**Comment by gregcaporaso** [comment]
Looks good, just one comment.

**Comment by jairideout** [comment]
Thanks for reviewing! I addressed your comment.

File: README.rst [hunk 2]
```diff
@@ -5,103 +5,67 @@
...
--  `An Introduction to Applied
-   Bioinformatics <http://caporasolab.us/An-Introduction-To-Applied-Bioinformatics/>`__
--  `tax2tree <https://github.com/biocore/tax2tree>`__
+- `QIIME <http://qiime.org/>`__
+- `Emperor <http://biocore.github.io/emperor/>`__
+- `An Introduction to Applied
+  Bioinformatics <http://readIAB.org>`__
+- `tax2tree <https://github.com/biocore/tax2tree>`__
+- `Qiita <http://qiita.microbio.me>`__
+- `ghost-tree <https://github.com/JTFouquier/ghost-tree>`__
```
Comment (ElDeveloper) [inline README.rst]: Although rather recent, Platypus-Conquistador is also making use:
https://github.com/biocore/Platypus-Conquistador

Comment (jairideout) [inline README.rst]: Thanks, added!

**Comment by Jorge-C** [comment]
:+1:

**Comment by gregcaporaso** [comment]
Looks like all comments have been addressed, merging. Thanks!

**Pull request merged**